src/sp/it/pl/util/access/fieldvalue/FileField.kt

```diff
@@ -110,16 +111,15 @@ private fun File.readBasicFileAttributes(): BasicFileAttributes? =
             null
         }
 
-private fun LocalDateTime.toFileTime() = FileTime.fromMillis(toInstant(ZoneOffset.UTC).toEpochMilli())
+private fun Long.toFileTime() = FileTime.fromMillis(this)
 
-private fun java.util.Date.toFileTime() = FileTime.fromMillis(time)
+private fun LocalDateTime.toFileTime() = toInstant(ZoneOffset.UTC).toEpochMilli().toFileTime()
 
 private fun File.readXmpTimeCreated(): FileTime? =
         try {
-            ImageMetadataReader
-                    .readMetadata(this)
-                    .getFirstDirectoryOfType(XmpDirectory::class.java)
-                    ?.getDate(XmpDirectory.TAG_CREATE_DATE)?.toFileTime()
+            ImageMetadataReader.readMetadata(this).getFirstDirectoryOfType(XmpDirectory::class.java)
+                    ?.xmpMeta?.getPropertyDate(Schema.XMP_PROPERTIES, "CreateDate")
```

Comment:
I'll test this, but for starters: is there no constant for the `"CreateDate"` value? This is pretty bad.

Reply:
Ok